docs/2024/puzzles/day03.md

```diff
@@ -6,6 +6,120 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 https://adventofcode.com/2024/day/3
 
+## Solution Summary
+
+1. Write a regex that will find a string which looks like `"mul(a,b)"` where `a` and `b` are integers.
+2. **Part 1**:
+   - Run the regex over the input string and find all matches.
+   - For each match extract `a` and `b` and multiply them.
+   - Sum all the partial results.
+3. **Part 2**:
+   - Write a regex that will look for either the same string as in **Part 1** or `"do()"` or `"don't()"`.
+   - Run the regex over the input string and find all matches.
+   - Fold over the matches and use the regex from **Part 1** to multiply and sum only from the "highlighted" parts of the input string.
+
+## Part 1
+
+This task plays very well to Scala strengths: pattern matching, and the standard collections library. But first we need to learn a bit of regular expressions. Personally, I believe everyone should learn them. They're scary only at the beginning, and then they because a very valuable tool in your pocket.
+
+The shopkeeper at the North Pole Toboggan Rental Shop asks us to go through the input string, find all substrings that look like calls to a multiplication method (`"mul(a,b)`), perform the multiplications, and sum them together. To find such substrings we will construct a regular expression which will look for the following:
+1. Each substring starts with a text `mul(`. In the regex form we need to escape the parenthesis, so this will be `mul\(`.
+2. The text is followed by the first positive integer. Since it's an integer and it's positive we don't need to look for floating points or minuses, we can just look for sequences of ciphers. In regex, one cipher is denoted by `\d`, and since we look for a sequence of one or more ciphers, it's `\d+`.
+3. After the first number there should be a comma, and in regex that again needs to be escaped, so `\,`.
+4. Then there should be a second number, so agian `\d+`.
+5. And the substring should finish with the closing parentheses, escaped again, so `\)`.
+That's it. Our whole regex looks like this: 
+```scala
+val mulPattern: Regex = """mul\((\d+)\,(\d+)\)""".r
```

Comment:
`,` doesn't need to be quoted:
```suggestion
val mulPattern: Regex = """mul\((\d+),(\d+)\)""".r
```
Also, have you considered using `java.util.regex.Pattern.quote` or the `\Q...\E` escapes? They might not do much good for `mul`, but for `do()` and `don't()` that would probably make it easier to read.

Reply:
I made one more commit, removing the part about escaping commas.